src/NuGet.Core/NuGet.Frameworks/NuGetFramework.cs

```diff
@@ -120,10 +188,18 @@ public string GetDotNetFrameworkName(IFrameworkNameProvider mappings)
 
                 parts.Add(string.Format(CultureInfo.InvariantCulture, "Version=v{0}", GetDisplayVersion(framework.Version)));
 
-                if (!string.IsNullOrEmpty(framework.Profile))
+                if (!IsNet5Era && !string.IsNullOrEmpty(framework.Profile))
                 {
                     parts.Add(string.Format(CultureInfo.InvariantCulture, "Profile={0}", framework.Profile));
                 }
+                if (IsNet5Era && !string.IsNullOrEmpty(framework.Platform))
+                {
+                    parts.Add(string.Format(CultureInfo.InvariantCulture, "Platform={0}", framework.Platform));
+                    if (framework.PlatformVersion != FrameworkConstants.EmptyVersion)
+                    {
+                        parts.Add(string.Format(CultureInfo.InvariantCulture, "PlatformVersion=v{0}", GetDisplayVersion(framework.PlatformVersion)));
```

Comment:
did we vet this with @terrajobst @dsplaisted @mhutch?

Reply:
I was thinking that the platform information would not round-trip into the .NET TargetFrameworkMoniker.  Is there any need to represent the platform info there?

Reply:
This is what we write in the nuspec. 
The platform & version are a part of the more complete framework identifier.

Reply:
If it just goes in the nuspec then it's more or less an implementation detail of NuGet.  

- The string does surface in some cases (I think NuGet.org uses it to some degree)
- We had talked about changing the format of this information in the nuspec to use the short framework name.  This would avoid a net5.0 package still showing .NETCoreApp as the target framework identifier
- Is the nuspec the only place this shows up?  Are there other (possibly non-NuGet) consumers of the string in this format?

Reply:
Nope, this logic hasn't been vetted. This might be worth discussing, it seems?

Reply:
We write it in the nuspec.
We show it in the PM UI, nuget.org etc.

This is the primary "pivot" for packages though. 

So everyone should be able to see a dependency group is specific to the IOS platform.

Reply:
I've suggested in the design issue to use `TargetPlatform*` properties separately without linking the `TargetFramework*` property, in which case nuget adds `NuGetPlatform` as opposed to `NuGetFramework` and similar classes/property names. If that is too much then use profiles, its a lot simpler than adding `Platform` concept to nuget.

Reply:
This logic has been updated to follow https://github.com/dotnet/designs/pull/114